### PR TITLE
read packets with the maximum packet size

### LINKED ISSUE #467: too small MaxPacketSize for receiving packets
We're currently defining a `MaxPacketSize` of 1350 bytes.

According to the Wire Spec, the maximum packet size is 1350 bytes for IPv6 and 1370 bytes for IPv4. That means that for sending packets, we're complying with the spec. However, we're only allocating `MaxPacketSize` bytes for reading packets in the [server.go:L86](https://github.com/lucas-clemente/quic-go/blob/eff9c6f90649c8aa63b155b1143237c6120fba96/server.go#L86), which means that we're not able to read packets using the full 1370 bytes allowed on an IPv4 connection.

### PR DESCRIPTION
**Pull request by marten-seemann**
fixes #467

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func (c *client) listen() {
 }
 
 func (c *client) handlePacket(remoteAddr net.Addr, packet []byte) error {
-	if protocol.ByteCount(len(packet)) > protocol.MaxPacketSize {
+	if protocol.ByteCount(len(packet)) > protocol.MaxReceivePacketSize {
```
Comment (lucas-clemente) [inline client.go]: Can this actually ever happen if we only read packets of a certain size? Can you look into whether there's some kind of signal from the kernel when packets got truncated?

File: server.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ func (s *server) Addr() net.Addr {
 }
 
 func (s *server) handlePacket(pconn net.PacketConn, remoteAddr net.Addr, packet []byte) error {
-	if protocol.ByteCount(len(packet)) > protocol.MaxPacketSize {
+	if protocol.ByteCount(len(packet)) > protocol.MaxReceivePacketSize {
```
Comment (lucas-clemente) [inline server.go]: see above

File: client.go (hunk 1 above)
Comment (marten-seemann) [inline client.go]: It cannot, and I already opened a new issue for that: #468. Afaik there's no signal, so we'll have to allocate 1 more byte in the packet buffer if we want to perform this check.

**Comment by lucas-clemente** [comment]
lgtm (the review button vanished).

**Pull request merged**